components/Shared/Button/BaseButton.js

```diff
@@ -1,17 +1,20 @@
 import styled from 'styled-components'
-import { space, layout, borderRadius, flexbox } from 'styled-system'
+import { space, layout, borderRadius, flexbox, variant } from 'styled-system'
 
 export default styled.button`
+  variant {
+    scale: 'buttons';
+  }
   cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
   background-color: ${props =>
     props.disabled
       ? props.theme.colors.status.inactive
-      : props.theme.colors.buttons[props.buttonStyle].background};
+      : props.theme.colors.buttons[props.variant].background};
   border-color: ${props =>
     props.disabled
       ? props.theme.colors.status.inactive
-      : props.theme.colors.buttons[props.buttonStyle].borderColor};
-  color: ${props => props.theme.colors.buttons[props.buttonStyle].color};
+      : props.theme.colors.buttons[props.variant].borderColor};
+  color: ${props => props.theme.colors.buttons[props.variant].color};
```

Comment:
Instead of declaring these conditionals here, should we now create a `disabled` button variant, and pass these conditional props at the point they're being rendered?

Cons: Might create a bunch of redundant code as we'd declare this prop relationship (disabled = disabled button state) across the app rather than inside this `const`

Reply:
I don't think so because we might eventually want different disabled looks based on the button's variant